### PR TITLE
WT-12709 Fix timestamp of reinstantiated deletes being overwritten at commit

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_truncate24.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        uri = 'table:truncate24'
+        ds = SimpleDataSet(self, uri, 1000000, key_format=self.key_format, value_format=self.value_format)
+        ds.populate()
+
+        # Reopen the connection to flush the cache so we can fast-truncate.
+        self.reopen_conn()
+
+        cursor = self.session.open_cursor(uri)
+
+        # Truncate the data at timestamp 10 but commit at 30.
```
Comment (pmacko86) [inline test/suite/test_truncate24.py:54]: Nitpick: This transaction actually commits at timestamp 20.

File: test/suite/test_truncate24.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        self.session.begin_transaction()
+        self.session.timestamp_transaction("commit_timestamp=" + self.timestamp_str(10))
+        self.session.truncate(uri, None, None, None)
+        # Reinstantiate the deleted data
```
Comment (pmacko86) [inline test/suite/test_truncate24.py:58]: What does "reinstantiate" mean in this context? It may be worth explaining a bit more in the comment...

**Review by pmacko86 — APPROVED** [review]
LGTM, with a few minor comment questions.

File: test/suite/test_truncate24.py (hunk 2 above)
Comment (sueloverso) [reply test/suite/test_truncate24.py:58]: Minor nit: Please end all comment sentences with a period.

**Review by sueloverso — APPROVED** [review]

File: test/suite/test_truncate24.py (hunk 1 above)
Comment (quchenhao) [reply test/suite/test_truncate24.py]: fixed

File: test/suite/test_truncate24.py (hunk 2 above)
Comment (quchenhao) [reply test/suite/test_truncate24.py]: done

Comment (quchenhao) [reply test/suite/test_truncate24.py]: @pmacko86 We reinstantiate the deleted pages back into memory if there is a reader for the page by putting tombstones on the keys. These tombstones need to be tracked when the truncated is committed.

**Pull request merged**